src/commands/deploy.ts

```diff
@@ -180,17 +175,31 @@ export default class Deploy extends Command {
   protected async runRemote() {
     const { args, flags } = this.parse(Deploy);
 
-    if (!args.environment_config) {
+    if (!args.environment_config_or_component) {
       throw new EnvConfigRequiredError();
     }
 
-    const env_config_path = path.resolve(untildify(args.environment_config));
-    // Validate env config
-    const env_config = await EnvironmentConfigBuilder.buildFromPath(env_config_path);
-    for (const [ck, cv] of Object.entries(env_config.getComponents())) {
-      if (cv.getExtends()?.startsWith('file:')) {
-        this.error(`Cannot deploy component remotely with file extends: ${ck}: ${cv.getExtends()}`);
+    let env_config: EnvironmentConfig;
+
+    let merge: boolean;
+    if (args.environment_config_or_component.includes(':')) {
+      const [name, tag] = args.environment_config_or_component.split(':');
```

Comment:
Should we error check this for at least an amount of split parts or something? just the condition to get into this code feels very fragile